From c501310bb2614f9a25b8f9628473e0e658b71e88 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:41 -0400
Subject: [misc] mei: mei_release: drop redundant check if cb is NULL

Message-id: <1399041470-21330-9-git-send-email-prarit@redhat.com>
Patchwork-id: 79297
O-Subject: [RHEL7.1 PATCH BZ 1088953 08/77] mei: mei_release: drop redundant check if cb is NULL
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit a9c8a17aea8ced19598b388db618e9094f3c1245
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Sun Sep 15 18:11:06 2013 +0300

    mei: mei_release: drop redundant check if cb is NULL

    mei_io_cb_free follows kfree design and check for NULL internally

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/main.c b/drivers/misc/mei/main.c
index b2221a3..067bb6d 100644
--- a/drivers/misc/mei/main.c
+++ b/drivers/misc/mei/main.c
@@ -165,10 +165,7 @@ static int mei_release(struct inode *inode, struct file *file)
 
  file->private_data = NULL;
 
- if (cb) {
-  mei_io_cb_free(cb);
-  cb = NULL;
- }
+ mei_io_cb_free(cb);
 
  kfree(cl);
 out:
-- 
1.7.1